cli/tools/test/channel.rs

```diff
@@ -253,7 +265,7 @@ impl TestEventSenderFactory {
         loop {
           tokio::select! {
             _ = test_stdout.pipe(), if test_stdout.is_alive() => {},
-            _ = test_stderr.pipe(), if test_stdout.is_alive() => {},
+            _ = test_stderr.pipe(), if test_stderr.is_alive() => {},
```

Comment:
This typo here meant that we were polling test_stderr in a hot loop if it finished, never getting to the sync receiver